git-ftp

```diff
@@ -412,14 +412,34 @@ fire_upload_buffer() {
 	if [ ! -f "$TMP_CURL_UPLOAD_FILE" ]; then
 		return 0
 	fi
-	print_info "Uploading ..."
+	local BUFFERED_ITEMS="$1"
+	print_info "Uploading $BUFFERED_ITEMS files..."
+	REMOTE_CMD_OPTIONS=""
 	set_default_curl_options
 	CURL_ARGS+=(--ftp-create-dirs)
 	CURL_ARGS+=(-K "$TMP_CURL_UPLOAD_FILE")
-	curl "${CURL_ARGS[@]}"
+
+	if [ $VERBOSE -eq 0 ]; then
+		CURL_ARGS+=(--progress-bar)
+		curl "${CURL_ARGS[@]}" 2> >(print_progress $BUFFERED_ITEMS) > /dev/null; test ${PIPESTATUS[0]} -eq 0
+	else
+		curl "${CURL_ARGS[@]}"
+	fi
+
 	check_exit_status "Could not upload files." "$ERROR_UPLOAD"
 }
 
+print_progress() {
+	local BUFFERED_ITEMS="$1"
+	local UPLOADED=0
+	while IFS= read -r line
```

Comment:
You just take each line as the output of a successful progress. What if there are error messages in between? For example when it can't upload one file, curl will print an error and then continue. Would that confuse your counting?